### PR TITLE
Misc Fixes w/ Removal of `support/http/server` package

### PR DESCRIPTION
**Pull request by nullstyle**
This is a grab bag PR, cherry-picked from ongoing work to improve friendbot and transaction submission in horizon.

Changes:
- Remove `support/http/server` package
- Moves more code to the `clients/stellarcore` package
- Moves logging middleware out of `support/log`, simplifying dependencies
- Move federation from goji to chi for muxing
- Added "Mux" helpers to `support/http` to get our common configurations together

### CODE REVIEW

File: support/config/main.go [hunk 1]
```diff
@@ -12,6 +12,12 @@ import (
 	"github.com/stellar/go/support/errors"
 )
 
+// TLS represents a common configuration snippet for configuring TLS in a server process
+type TLS struct {
```
Comment (nikhilsaraf) [inline support/config/main.go]: 👍 for extracting this out

File: support/http/logging_middleware.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+func logEndOfRequest(
+	ctx context.Context,
+	duration time.Duration,
+	mw mutil.WriterProxy,
+) {
+	log.Ctx(ctx).WithFields(log.F{
+		"subsys":   "http",
+		"status":   mw.Status(),
+		"bytes":    mw.BytesWritten(),
+		"duration": duration,
```
Comment (bartekn) [inline support/http/logging_middleware.go]: What do you think about adding `method` and `path` here too? Would be really helpful for per endpoint `duration` analysis.

Also `req` would be great to find all log entries within a single HTTP request. I found it really helpful when analysing invite system logs in the past.

I know this PR is mostly moving things around so can be done in a separate PR.

Comment (nullstyle) [reply support/http/logging_middleware.go]: > Also req would be great to find all log entries within a single HTTP request. I found it really helpful when analysing invite system logs in the past.

Thanks for the catch.  Leaving `req` out is definitely a bug.

> What do you think about adding method and path here too? Would be really helpful for per endpoint duration analysis.

Before we duplicate the info, we should try building a system that matches the start and the finish log lines through `req`, giving us access to the path and method.  make sense?

Comment (bartekn) [reply support/http/logging_middleware.go]: > Before we duplicate the info, we should try building a system that matches the start and the finish log lines through `req`, giving us access to the path and method. make sense?

In my experience using tools like Loggly it's really hard (or even impossible) to find matching requests. But I'm OK with not doing this now.

Comment (nullstyle) [reply support/http/logging_middleware.go]: @bartekn could I get another review?  Thanks :)

File: support/http/logging_middleware.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+func logStartOfRequest(
+	ctx context.Context,
+	r *stdhttp.Request,
+) {
+	log.Ctx(ctx).WithFields(log.F{
+		"subsys": "http",
+		"path":   r.URL.String(),
+		"method": r.Method,
+		"ip":     r.RemoteAddr,
+		"host":   r.Host,
```
[the remark this answers is not in the document]
Comment (nullstyle) [reply support/http/logging_middleware.go]: misc_fixes_w__removal_of_ support_http_server _package_by_nullstyle_ _pull_request__395_ _stellar_go

`req` should appear on _every_ log line emitted from a request by embedding a new logger scoped with `req` to the request context.  The change screenshotted is the start of that. It still needs some work to get properly set on the other lines emitted from during a request, but I'll work on that separately.

**Pull request merged**